server/api.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/msteams"
+)
+
+type API struct {
+	p      *Plugin
+	router *mux.Router
+}
+
+type Activities struct {
+	Value []msteams.Activity
+}
+
+func NewAPI(p *Plugin) *API {
+	router := mux.NewRouter()
+	api := &API{p: p, router: router}
+
+	router.HandleFunc("/avatar/{userId:.*}", api.getAvatar).Methods("GET")
+	router.HandleFunc("/", api.processMessage).Methods("POST")
+
+	return api
+}
+
+func (a *API) getAvatar(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	userID := params["userId"]
+	photo, appErr := a.p.API.KVGet("avatar_" + userID)
```

Comment:
Do we have a way to update this value when the user's avatar changes in MSTeams?

Reply:
It is cached for 5 minutes, (i set it with `KVSetWithExpiry`)